security/tpm/src/tpm.ts

```diff
@@ -3,22 +3,423 @@
 
 'use strict';
 import { errors } from 'azure-iot-common';
+import * as machina from 'machina';
+import * as tss from 'tss.js';
+import { Tpm, TPM_HANDLE, TPM_ALG_ID, TPM_RC, TPM_PT, TPMA_OBJECT, TPMT_PUBLIC, TPM2B_PRIVATE } from 'tss.js';
+import * as crypto from 'crypto';
+import base32Encode = require('base32-encode');
+
+import * as dbg from 'debug';
+
+const debug = dbg('azure-iot-security-tpm:TpmSecurityClient');
+
+
+const aes128SymDef = new tss.TPMT_SYM_DEF_OBJECT(TPM_ALG_ID.AES, 128, TPM_ALG_ID.CFB);
+
+const ekPersHandle: TPM_HANDLE = new TPM_HANDLE(0x81010001);
+const srkPersHandle: TPM_HANDLE = new TPM_HANDLE(0x81000001);
+const idKeyPersHandle: TPM_HANDLE = new TPM_HANDLE(0x81000100);
+
+// Template of the Endorsement Key
+const ekTemplate = new TPMT_PUBLIC(TPM_ALG_ID.SHA256,
+  TPMA_OBJECT.restricted | TPMA_OBJECT.decrypt | TPMA_OBJECT.fixedTPM | TPMA_OBJECT.fixedParent | TPMA_OBJECT.adminWithPolicy | TPMA_OBJECT.sensitiveDataOrigin,
+  new Buffer('837197674484b3f81a90cc8d46a5d724fd52d76e06520b64f2a1da1b331469aa', 'hex'),
+  new tss.TPMS_RSA_PARMS(aes128SymDef, new tss.TPMS_NULL_ASYM_SCHEME(), 2048, 0),
+  new tss.TPM2B_PUBLIC_KEY_RSA());
+
+// Template of the Storage Root Key
+const srkTemplate = new TPMT_PUBLIC(TPM_ALG_ID.SHA256,
+  TPMA_OBJECT.restricted | TPMA_OBJECT.decrypt | TPMA_OBJECT.fixedTPM | TPMA_OBJECT.fixedParent | TPMA_OBJECT.noDA | TPMA_OBJECT.userWithAuth | TPMA_OBJECT.sensitiveDataOrigin,
+  null,
+  new tss.TPMS_RSA_PARMS(aes128SymDef, new tss.TPMS_NULL_ASYM_SCHEME(), 2048, 0),
+  new tss.TPM2B_PUBLIC_KEY_RSA());
+
+
+interface PersistentKeyInformation {
+  /**
+   * Human readable name for debugging/logging purposes only
+   */
+  name: string;
+
+  /**
+   * The TPM hierarachy, to which the key belongs
+   * @note All the keys are persisted in the owner hierarchy disregarding their mother one
```

Comment:
>y disregarding their mother one [](start = 59, length = 31)

the "disregarding..." phrase makes no sense to me.

Reply:
This structure is GONE!

---
In reply to: [159308125](https://github.com/Azure/azure-iot-sdk-node/pull/185#discussion_r159308125) [](ancestors = 159308125)